config/schemas/config.yaml

```diff
@@ -1693,6 +1693,13 @@ properties:
               `RetentionDays` defines how old a backup should be before deleting it.
             default: 7
             type: number
+          schedule:
+            title: Schedule for backup job
+            description: |-
+              `schedule` defines when the backup job for harbor will run.
+              This should be set to run shortly after velero backups in wc, in order to ensure that images needed for velero backups are backed up in harbor.
+            default: "30 0 * * *"
```

Comment:
It could be nice to have a pattern for this in the schema

Reply:
I did manage to find this pattern that seems to work in all general cases, but there might be some edge cases where it fails. But it is very complicated.

```
^((?<![\d\-\*])((\*\/)?([0-5]?[0-9])((\,|\-|\/)([0-5]?[0-9]))*|\*)[^\S\r\n]+((\*\/)?((2[0-3]|1[0-9]|[0-9]|00))((\,|\-|\/)(2[0-3]|1[0-9]|[0-9]|00))*|\*)[^\S\r\n]+((\*\/)?([1-9]|[12][0-9]|3[01])((\,|\-|\/)([1-9]|[12][0-9]|3[01]))*|\*)[^\S\r\n]+((\*\/)?([1-9]|1[0-2])((\,|\-|\/)([1-9]|1[0-2]))*|\*|(jan|feb|mar|apr|may|jun|jul|aug|sep|oct|nov|dec))[^\S\r\n]+((\*\/)?[0-6]((\,|\-|\/)[0-6])*|\*|00|(sun|mon|tue|wed|thu|fri|sat)))$|^@(annually|yearly|monthly|weekly|daily|hourly|reboot)$
````

Is this something we want, or is it just confusing to add this?

Reply:
The Kubernetes spec just has this:
```
        "schedule": {
          "description": "The schedule in Cron format, see https://en.wikipedia.org/wiki/Cron.",
          "type": "string"
        },
```

Reply:
If we really want something this seems less complicated:
`/(@(annually|yearly|monthly|weekly|daily|hourly|reboot))|(@every (\d+(ns|us|µs|ms|s|m|h))+)|((((\d+,)+\d+|(\d+(\/|-)\d+)|\d+|\*) ?){5,7})/`
Source: https://stackoverflow.com/questions/14203122/create-a-regular-expression-for-cron-statement

Reply:
Added a link now to all the config options I could find that uses a cron format.

Reply:
Going with link only for now?

Reply:
That's what I was thinking. But I'm open to the option of using the complicated regex as well.

Reply:
I think I would like to have the regex to get some config validation before the k8s deploy stage. I don't really have a strong opinion on this though!

Reply:
I have now added a regex pattern. I had to exclude "(?<![\d\-\*])" from the pattern I mentioned above because negative lookahead is not supported in the schema. But I think that is fine.

Here is some validation of the regex pattern that I used: https://regex101.com/r/l56Too/2